test/ScanDependencies/test_clang_gmodules.swift

```diff
@@ -1,6 +1,6 @@
 // RUN: %empty-directory(%t)
 // RUN: %empty-directory(%t/module-cache)
-// RUN: %target-swift-frontend -scan-dependencies -module-cache-path %t/module-cache %s -o %t/deps.json -I %S/Inputs/CHeaders -I %S/Inputs/Swift -target %target-cpu-apple-macosx10.14
+// RUN: %target-swift-frontend -scan-dependencies -module-cache-path %t/module-cache %s -o %t/deps.json -I %S/Inputs/CHeaders -I %S/Inputs/Swift -target %target-cpu-apple-macosx10.14 -disable-implicit-concurrency-module-import -disable-implicit-string-processing-module-import
```

Comment:
Does this test mean it usually triggered from implicit imported modules? And implicit module cannot trigger cross import?

Reply:
No, these implicit imports do not actually result in any cross-import overlays. The way we noticed an issue here is that implicitly-imported Concurrency module was not able to be resolved for a given target platform. And that caused there to be a hanging unresolved import, which caused the code this PR is changing to crash. 

So I wanted to both address this issue and also fix the test to not have a possibility of this missing module error which is fully unrelated to what it is meant to be testing.

Reply:
Is it possible to write a test for the failing case previously?